src/Libraries/Microsoft.Extensions.AI/ChatCompletion/AnonymousChatClient.cs

```diff
@@ -0,0 +1,242 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+#pragma warning disable CA1068 // CancellationToken parameters must come last
+#pragma warning disable VSTHRD003 // Avoid awaiting foreign Tasks
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>A delegating chat client that wraps an inner client with implementations provided by delegates.</summary>
+public sealed class AnonymousChatClient : DelegatingChatClient
+{
+    /// <summary>Services associated with the chat client.</summary>
+    private readonly IServiceProvider? _services;
+
+    /// <summary>The delegate to use as the implementation of <see cref="CompleteAsync"/>.</summary>
+    private readonly Func<Arguments, Task<ChatCompletion>>? _completeFunc;
+
+    /// <summary>The delegate to use as the implementation of <see cref="CompleteStreamingAsync"/>.</summary>
+    /// <remarks>
+    /// When non-<see langword="null"/>, this delegate is used as the implementation of <see cref="CompleteStreamingAsync"/> and
+    /// will be invoked with the same arguments as the method itself, along with a reference to the inner client.
+    /// When <see langword="null"/>, <see cref="CompleteStreamingAsync"/> will delegate directly to the inner client.
+    /// </remarks>
+    private readonly Func<Arguments, IAsyncEnumerable<StreamingChatCompletionUpdate>>? _completeStreamingFunc;
+
+    /// <summary>The delegate to use as the implementation of both <see cref="CompleteAsync"/> and <see cref="CompleteStreamingAsync"/>.</summary>
+    private readonly Func<Arguments, Func<IList<ChatMessage>, ChatOptions?, CancellationToken, Task>, Task>? _sharedFunc;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AnonymousChatClient"/> class.
+    /// </summary>
+    /// <param name="innerClient">The inner generator.</param>
```

Comment:
Presumably this was copied over by the equivalent embedding generator?
```suggestion
    /// <param name="innerClient">The inner client.</param>
```